contracts/TokenFactory.sol

```diff
@@ -326,18 +358,71 @@ contract TokenFactory is ReentrancyGuard, Ownable {
         }
     }
 
-    function burnTokenAndMintWinner(
+    function publishToUniswap (
         address tokenAddress
     ) external nonReentrant {
         uint256 _competitionId = competitionIds[tokenAddress];
+
+        require(_competitionId != currentCompetitionId, 'The competition is still active');
+
+        address winnerToken = getWinnerByCompetitionId(_competitionId);
+
+        require(winnerToken == tokenAddress, 'Token address not winner');
+
+        uint256 totalCollateral = getCollateralByCompetitionId(_competitionId);
+        WETH.deposit{value: totalCollateral}();
+
+        // calulate winner token amount
+        uint256 contributionWithoutFee = (totalCollateral * FEE_DENOMINATOR) /
```

Comment:
fee was already assessed before being added into collateral. It should not be deducted again here